### PR TITLE
React DOM: Add support for `hidden="until-found"`

### PR DESCRIPTION
**Pull request by eps1lon**
## Summary

Closes https://github.com/facebook/react/issues/24740

Adds support for the experimental `hidden="until-found"` as well as the correspondig TODO `onBeforeMatch` event handler.

Keep in mind that these new browser APIs are only implemented in Chrome and Edge at the moment. Safari and Firefox have no support for this API yet (see [`hidden` browser compatibility table](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden#browser_compatibility). 

I haven't checked if we can polyfill the API or maybe we shouldn't even do that? Otherwise landing this would take more time and block experimentation in UIs targetting Chromium exclusively

## How did you test this change?

- [x] Attribute fixture (commited changes with non-experimental build since this is the build we were using before)
- [x] [Codesandbox from repro](https://codesandbox.io/s/lingering-river-4jor1y) will contain matching behavior of DOM and React: https://codesandbox.io/s/lucid-benz-u2oksw

### CODE REVIEW

File: fixtures/attribute-behavior/AttributeTableSnapshot.md [hunk 1]
```diff
@@ -4827,20 +4827,20 @@
 | Test Case | Flags | Result |
 | --- | --- | --- |
 | `hidden=(string)`| (changed)| `<boolean: true>` |
-| `hidden=(empty string)`| (initial)| `<boolean: false>` |
+| `hidden=(empty string)`| (changed)| `<boolean: true>` |
```
Comment (eps1lon) [inline fixtures/attribute-behavior/AttributeTableSnapshot.md:4830]: Similar issue as https://github.com/facebook/react/pull/24730: Should this be behind a flag?

**Comment by eps1lon** [comment]
We'll also need to look into support for `beforematch` events and we need to decide if we're ok with experimental APIs that are only supported by a single browser.

**Comment by chaance** [comment]
@eps1lon Any reason not to reopen this one?

**Comment by mj12albert** [comment]
> experimental APIs that are only supported by a single browser.

Firefox supports `hidden="until-found"` now: https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Releases/139#html

**Comment by PeterDekkers** [comment]
This is also part of the targets for [Interop 2025](https://wpt.fyi/results/html/editing/the-hidden-attribute?label=master&label=experimental&aligned&view=interop&q=label%3Ainterop-2025-details), so should see wide browser support later this year.

**Pull request closed without merging**